src/shared/components/palette/document/write-document-field.component.ts

```diff
@@ -136,35 +145,48 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
       });
     } else {
 
-      this.selectedFile = null;
-      this.uploadedDocument.get('document_url').setValue(null);
-      this.uploadedDocument.get('document_binary_url').setValue(null);
-      this.uploadedDocument.get('document_filename').setValue(null);
-
+      this.setComponentToInitialValues();
```

Comment:
there is a change of behaviour here. The method `setComponentToInitialValues` executes only if this is a mandatory component. But the code above was executing always. Is this what we want?